### DISCUSSION TITLE
Data segmentation

### DISCUSSION THREAD

**Discussion by aelhadee** [named-data/python-ndn#50] [Q&A] [answered]
Hello, 
I have a quick question, has the data segmentation for data larger than 8800 bytes been implemented in this repo or it needs to be incorporated manually?

Thanks,
AE

**Comment by zjkmxy** [comment]
Python-ndn does not do segmentation autmatically.
Actually, python-ndn is not aware of this limit, because this is required only by NFD. NFD will drop your Data packet if it is larger than 8800B. This is applied to Data packets after fragment reassembly, so the library generally cannot help you do anything.
I recommend you to segment your data payload in a size of about 4~6kB.

**Reply by aelhadee** [reply]
Hello @zjkmxy, I have tried data segmentation with python-ndn at both the sender and receiver and was not successful. Is there a sample segmentation code that I can refer to at both devices?

**Reply by zjkmxy** [reply]
For the producer, you can refer to [putchunks.py](https://github.com/named-data/python-ndn/blob/master/examples/putchunks.py).
For the producer, you can refer to [catchunks.py](https://github.com/named-data/python-ndn/blob/master/examples/catchunks.py).

You can also refer to [rdrnode.py](https://github.com/named-data/python-ndn/blob/master/examples/rdrnode.py), which implements the [RDR protocol](https://redmine.named-data.net/projects/ndn-tlv/wiki/RDR) and acts as both producer and consumer. However, this example uses NTSchema API, which is experimental and may change frequently.

**Comment by aelhadee** [comment, marked as answer]
These really helped! Thank you!

> For the producer, you can refer to [putchunks.py](https://github.com/named-data/python-ndn/blob/master/examples/putchunks.py).
> For the producer, you can refer to [catchunks.py](https://github.com/named-data/python-ndn/blob/master/examples/catchunks.py).

**Comment by aelhadee** [comment]
Hi @zjkmxy, I have tried to use putchunks to send fresh data by using data = os.urandom(32000) so it would send new data every time there is a new interest from catchunks, but it didn't seem to work. It seems like the same data is sent over cnt times. Any idea why that is happening?

**Reply by zjkmxy** [reply]
In NDN different data are supposed to have different names. If catchunks sends the same Interest again, it will hit NFD's cache instead of the producer.

**Reply by aelhadee** [reply]
So for some reason when the time stamp is added to the name in putchunks.py, the catchunks.py does not detect it and the interest times out. Is there a reason for that? It only works when I remove the timestamp and just use the name by itself in putchunks.py.
